kfs-core/src/main/java/edu/arizona/kfs/sys/batch/service/impl/BuildingImportServiceImpl.java

```diff
@@ -0,0 +1,529 @@
+			reportMsg += "BuildingAddressZipCode(" + building.getBuildingAddressZipCode() + "), ";
+			building.setBuildingAddressZipCode(archBuilding.getBuildingAddressZipCode());
+		}
+		if (archibusUsaCountryCode.equalsIgnoreCase(archBuilding.getBuildingAddressCountryCode()) && kfsUsCountryCode.equalsIgnoreCase(building.getBuildingAddressCountryCode())) {
+		}
+		else {
+			reportMsg += "BuildingAddressCountryCode(" + building.getBuildingAddressCountryCode() + "), " ; 
+			if (archibusUsaCountryCode.equalsIgnoreCase(archBuilding.getBuildingAddressCountryCode())) {
+				building.setBuildingAddressCountryCode(kfsUsCountryCode);
+			}
+			else {
+				building.setBuildingAddressCountryCode(archBuilding.getBuildingAddressCountryCode());
+			}
+		}
+		
+		if (!archBuilding.getBuildingName().equalsIgnoreCase(building.getBuildingName())) {
+			reportMsg += "BuildingName(" + building.getBuildingName() + "), ";
+			building.setBuildingName(archBuilding.getBuildingName());
+		}
+		
+		if (StringUtils.isNotBlank(buildingExt.getRouteCode())) {
+			kfsBuild = buildingExt.getRouteCode();
+		}
+		else {
+			kfsBuild = "";
+		}
+		if (KFSParameterKeyConstants.HYPHEN.equalsIgnoreCase(routeCodeValue)) {
+			archBuild = "";
+		}
+		else {
+			archBuild = routeCodeValue;
+		}
+		if (!archBuild.equalsIgnoreCase(kfsBuild)) {
+			if (!KFSParameterKeyConstants.HYPHEN.equalsIgnoreCase(routeCodeValue)) {
+				reportMsg += "RouteCode(" +buildingExt.getRouteCode() + "), ";
+				buildingExt.setRouteCode(archBuilding.getRouteCode());
+				buildingExt.setRouteCodeObj(routecode);
+			}
+			else {
+				buildingExt.setRouteCode(null);
+				buildingExt.setRouteCodeObj(null);
+			}
+		}
+		LOG.info("Updating (Name,Code,Campus,City,State,Zip,Country,Active,Address)(" + building.getBuildingName() +","+ building.getBuildingCode() +","+ building.getCampusCode() +","+ building.getBuildingAddressCityName() +","+ building.getBuildingAddressStateCode() +","+ building.getBuildingAddressZipCode() +","+ building.getBuildingAddressCountryCode() +","+ building.isActive() +","+ building.getBuildingStreetAddress() +")");
+		buildingExt.setCampusCode(campusCode);
+		buildingExt.setBuildingCode(building.getBuildingCode());
+		businessObjectService.save(building);
+		businessObjectService.save(buildingExt);
+		writetoReportOutputFile(outReportWriter, reportMsg);
+	}
+	
+	private void disableKFSBuildingsNotInArchibus(PrintWriter outReportWriter, Building kfsBuilding) {
+		ArchibusBuildings archibusBuilding;
+		String reportMsg;
+		
+		if (kfsBuilding.getBuildingCode().equalsIgnoreCase("201")) {
+			System.out.println("Debug buildingCode 201");
+		}
+		reportMsg = "";
+		LOG.debug("Processing Record: " + kfsBuilding.toString());
+		
+		//look for a valid active archibus building.  There is no campusCode in Archibus
+		Map<String, String> keys = new HashMap<String, String>();
+		keys.put("buildingCode", kfsBuilding.getBuildingCode());
+		keys.put("active", "A");
```

Comment:
You created the constant STRING_A for a method above, please use the constant here as well.